### ISSUE TITLE
Behavior of .where() breaking change

### ISSUE THREAD

**Issue by slavah** [Reactive-Extensions/RxJS#532]
Hi,
in the latest version .where() method has a different semantics in comparison to previous versions.

Current version example:
```javascript
var subj = new Rx.Subject();
var obs = subj.asObservable();

obs.select(function(v){return v+' ' + 'mutated'})
.where(function(){return true})
.subscribe(function(v){console.log(v)});

obs.select(function(v){return v+' ' + 'mutated'})
.subscribe(function(v){console.log(v)});

```

```javascript
subj.onNext("original")

original

original mutated

```
Is this change intended for "where()" method to send original message into observable chain?

Thanks

**Comment by mattpodwysocki** [comment]
@slavah fixed as per latest commit, see Issue #537

**Comment by slavah** [comment]
Thanks a lot. Yep, missed that its a duplicate issue.
